### PR TITLE
Better Tulane PR

### PR DESCRIPTION
**Pull request by outlawhayden**
Hey all -

Thanks for the feedback - sorry the first PR has been such a nightmare. We made a branch on outlawhayden that is the most recent version of eye-on-surveillance/sawt, _not ours_, and we're going to go through and add all of our changes into your most recent version and see if we can get it working locally in this new context. We're going to be adding a bunch of stuff to it today/tonight, so don't worry about anything in here for a little bit, we will totally ping you when we think we have a workable version.

As well, I'm opening this against your main branch only because I don't have permissions in the EOS one (which is good), but if at some point yall want to merge this into a new branch on your end we think that's probably the best course of action.

### CODE REVIEW

**Comment by outlawhayden** [comment]
@marvinmarnold @ayyubibrahimi when trying to run getanswer locally on your most recent build, we get the following error message. working on parsing it, but in the meantime, are the dvc files up to date to include the most recent documents that got added?

`curl -XPOST localhost:8080 -H "Content-Type: application/json" \\n    -d '{"query":"What is a proviso", "response_type":"general", "card_id":123456789}
<snip>
'File "/Users/culotta/Dropbox (Personal)/p/sawt/packages/googlecloud/functions/getanswer/inquirer.py", line 210, in get_general_summary_response_from_query
    docs = db.similarity_search(query, k=k)
  File "/Users/culotta/Dropbox (Personal)/p/virtual/sawt/lib/python3.9/site-packages/langchain/vectorstores/faiss.py", line 334, in similarity_search
    docs_and_scores = self.similarity_search_with_score(
  File "/Users/culotta/Dropbox (Personal)/p/virtual/sawt/lib/python3.9/site-packages/langchain/vectorstores/faiss.py", line 276, in similarity_search_with_score
    docs = self.similarity_search_with_score_by_vector(
  File "/Users/culotta/Dropbox (Personal)/p/virtual/sawt/lib/python3.9/site-packages/langchain/vectorstores/faiss.py", line 228, in similarity_search_with_score_by_vector
    raise ValueError(f"Could not find document for id {_id}, got {doc}")
ValueError: Could not find document for id 79d5d7de-aa2f-48af-a44b-0ab1b0549f30, got page_content='- Topic: Conditional Use Approval with Provisos\n- Summary: The council considered a Conditional Use subject to two provisos from the City Planning report, with the addition of six provisos, for a total of eight provisos. \n- Ordinance Number: M-21-118\n- Votes Summary:\n    Vote 1: Withdrawn - (N/A, N/A, N/A, N/A)\n- Decision/Key Actions: The substitute motion was withdrawn.\n- UID: fe7705ee-59a5-489b-a12b-0286167ec4dc\n\n- Topic: Regulations on Alcoholic Beverages\n- Summary: The council discussed regulations related to the sale and consumption of alcoholic beverages, including restrictions on signage, sale, and consumption, as well as hours of operation.\n- Ordinance Number: N/A\n- Votes Summary: N/A\n- Decision/Key Actions: The specific details and outcome of the discussion were not provided in the given text.\n- UID: fe7705ee-59a5-489b-a12b-0286167ec4dc' metadata={'source': '/Users/ayyub/Desktop/sawt/sawt/packages/backend/src/minutes_agendas_directory/Minutes 2021.json', 'seq_num': 767, 'title': None, 'page_number': None}`

**Comment by marvinmarnold** [comment]
Hey thanks for reworking this. I just added you as a contributor to Github if that helps unblock you at all. Anything you can do to get things stable by tonight would be great. I'd love to be able to spend tomorrow ensuring things are stable.

**Comment by outlawhayden** [comment]
Awesome - I agree. My goal is to get the getanswer setup working again by the end of the day which is critical, as I think getting the new answers will take overnight if I do it correctly, and then we can populate the db and make sure everything holds together.

We split up so that @mikafur32 is helping stitch together the frontend, and I'm going to take a swing at some of the backend stuff, and hopefully everything lines up nice and neat merging it in.

File: packages/googlecloud/functions/getanswer/main.py
Comment (outlawhayden) [inline packages/googlecloud/functions/getanswer/main.py:15]: commenting out gcloud functionality for local getanswer deployment - just saving it here for reference, but should be undone before full deployment

File: packages/googlecloud/functions/getanswer/main.py
Comment (outlawhayden) [inline packages/googlecloud/functions/getanswer/main.py:128]: commenting out gcloud functionality for local getanswer deployment, and changing what comes back so it's not posting to a db - just saving it here for reference, but should be undone before full deployment

**Comment by marvinmarnold** [comment]
@outlawhayden how are things looking? I just finished my day job and have some time to dive in.

**Comment by outlawhayden** [comment]
howdy - i managed to reconstruct some of my earlier progress, and now the new dataset. is getting processed now/overnight. to do so, i had to comment out some of the external posting and gcloud stuff, but the file i just posted is basically a companion script that reads in a csv of questions from a location, and then just handles the api calls all in a line and saves the output (really should be in bash but im lame). i think we're no longer in triage mode - this pr should just slide right on top of your current stuff on a new branch, and once we have the new data in our db we can do some testing to make sure everything plays nice tomorow

**Comment by marvinmarnold** [comment]
Great @outlawhayden. I'll approve once Vercel build is passing and you can merge. Then we can at least debug in staging tomorrow and hopefully land in prod tomorrow night.

@mikafur32 you should be able to do `yarn build` locally to debug build issues.

**Comment by mikafur32** [comment]
Perfect, am debugging the building now. Quickly double checking one of the UI features that's causing issues. 👍

File: packages/web/components/Rubric.tsx
Comment (marvinmarnold) [inline packages/web/components/Rubric.tsx:18]: Btw check out tailwindcss when you get a chance. Not a blocker, but fyi for future:

- https://tailwindcss.com/docs/width
- https://tailwindcss.com/docs/border-radius#rounded-corners
- https://tailwindcss.com/docs/margin

**Comment by mikafur32** [comment]
We are running into issues associated with typescript that we were able to bypass with the local run, that throw errors when building. So, we're reformatting a bunch of stuff to fit the typescript specs. It might be a bit later tonight/tomorrow morning before we are able to pass the vercel and fully merge.

**Comment by mikafur32** [comment]
@marvinmarnold do you know if Vercel auto-checks or does it have to be rerun? ie: that is that newest commit valid?

**Comment by marvinmarnold** [comment]
Not sure why it's not auto-running for you. Maybe you have to link your Github to Vercel? Anyways, I just reran it.

**Comment by mikafur32** [comment]
Also @marvinmarnold , on the topic of the slide/swipeable mechanism for the cards, if you want to take a look and try to implement that, absolutely go full steam ahead. I know you said you think it'd be pretty simple to do, but with finals and everything, I don't think we have the time tomorrow to implement that and verify that it works 100% before Thursday.

**Comment by mikafur32** [comment]
> Not sure why it's not auto-running for you. Maybe you have to link your Github to Vercel? Anyways, I just reran it.

Yeah, I realized my primary email was different than the one I gave you for my github, so I had to change it up.

**Comment by mikafur32** [comment]
> > Not sure why it's not auto-running for you. Maybe you have to link your Github to Vercel? Anyways, I just reran it.
> 
> Yeah, I realized my primary email was different than the one I gave you for my github, so I had to change it up.

@marvinmarnold, can you double check the verification on my email: msison1@tulane.edu and catherine's cbrooks2@tulane.edu

**Comment by marvinmarnold** [comment]
Invites sent

**Comment by marvinmarnold** [comment]
You have to click button in email accept the invitation btw. 
Screenshot 2023-12-05 at 11 25 40 PM

**Comment by marvinmarnold** [comment]
Going to bed. ttyl

**Review by marvinmarnold — APPROVED** [review]
Great job. Can you merge, and I'll start testing.

**Comment by mikafur32** [comment]
Awesome! So it passed, but we are still going to make a few performance changes before accepting the PR.
@marvinmarnold

**Pull request merged**
